compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/AnnotationDesugar.java

```diff
@@ -383,21 +383,22 @@ private BLangLambdaFunction defineAnnotations(Location pos, BLangClassDefinition
         BLangRecordLiteral mapLiteral = null;
         boolean isLocalObjectCtor = classDef.flagSet.contains(Flag.OBJECT_CTOR);
         if (!classDef.annAttachments.isEmpty()) {
-            function = defineFunction(pos, pkgID, owner);
-            mapLiteral = ASTBuilderUtil.createEmptyRecordLiteral(pos, symTable.mapType);
-            addAnnotsToLiteral(classDef.annAttachments, mapLiteral, pos, env, isLocalObjectCtor);
+            Location annotPos = classDef.annAttachments.get(0).pos;
+            function = defineFunction(annotPos, pkgID, owner);
+            mapLiteral = ASTBuilderUtil.createEmptyRecordLiteral(annotPos, symTable.mapType);
+            addAnnotsToLiteral(classDef.annAttachments, mapLiteral, annotPos, env, isLocalObjectCtor);
```

Comment:
Here also, IMO, passing the position of the first annotation attachment as the position is not correct.

Reply:
We can pass the default position here, but the position needs to be have the file name. Instead of create and pass a position as `new BLangDiagnosticLocation(pos.lineRange().fileName(), -1, -1, -1, -1, 0, 0)`, I used the first position of the annotation attachment.